### PR TITLE
Add centralized node replacement system for ScriptCanvas

### PR DESCRIPTION
**Pull request by onecent1101**
Signed-off-by: onecent1101 <liug@amazon.com>

## What does this PR do?
Context: Introduce centralized ScriptCanvas node replacement system, which developer can provide node replacement logic through json file anywhere in Gem/Project folder. (In this way, the node object doesn't need to manage replacement metadata, and we are able to extend node replacement not just for deprecation use case)

Phase 1: Add the system, and make system support method node replacement. (Utilizing .setreg as the node replacement file which is good enough for current use case, later if we want to extend editor runtime modification we should create specific asset file and file watcher)

Open to any suggestion about json file format and naming.

Example:
You have two functions, and old function should be replaced by new function
```
<Library
    ...
    Namespace="TestFunctions"
    Category="TestFunction">

    <Function Name="OldTestFunction">

    <Function Name="NewTestFunction">
</Library>
```
You can provide the replacement metadata in following json format
```
{
    "Amazon": {
        "ScriptCanvas": {
            "NodeReplacement": [
                {
                    "OldNode" : {
                        "Uuid": "E42861BD-1956-45AE-8DD7-CCFC1E3E5ACF",
                        "Method": "TestFunctions_OldTestFunction"
                    },
                    "NewNode" : {
                        "Uuid": "E42861BD-1956-45AE-8DD7-CCFC1E3E5ACF",
                        "Method": "TestFunctions_NewTestFunction"
                    }
                }
            ]
        }
    }
}
```

## How was this PR tested?
Tested manually make sure target node can be replaced correctly

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Components/NodeReplacementSystem.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <AzCore/Math/Uuid.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/string/string.h>
+
+#include <ScriptCanvas/Core/Node.h>
+#include <Editor/Include/ScriptCanvas/Bus/NodeReplacementBus.h>
+
+namespace ScriptCanvasEditor
+{
+    static constexpr const char NodeReplacementRootPath[] = "/Amazon/ScriptCanvas/NodeReplacement";
```
[the remark this answers is not in the document]
Comment (pollend) [reply Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Components/NodeReplacementSystem.h:20]: umm, how do we want to migrate the rest of the settings?

[the remark this answers is not in the document]
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Components/NodeReplacementSystem.h]: @pollend Would you mind clarifying a bit about your question?

[the remark this answers is not in the document]
Comment (pollend) [reply Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Components/NodeReplacementSystem.h]: is there any strategy with the rest of the settings would this just include only new settings?

[the remark this answers is not in the document]
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Components/NodeReplacementSystem.h]: Phase 1: Move internal maintained node replacement to a centralized external system, and migrate method node replacement functionality
Phase 2: Refactor replacement logic, and migrate nodeable node replacement functionality
After that, the system should be straightforward to extend, and we will keep adding more supports based on the requirement from community.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**